File: rockfast-ui/src/models/Todo.js
export default class Todo {
    constructor(id, name, dateCreated, complete, dateCompleted, userId) {
      this.id = id;
      this.name = name;
      this.dateCreated = dateCreated;
      this.complete = complete;
      this.dateCompleted = dateCompleted;
      this.userId = userId;
    }
  }
  

File: Rockfast.WebAPI/Rockfast.ApiDatabase/DomainModels/Todo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Rockfast.ApiDatabase.DomainModels
{
    public class Todo
    {
        [Key]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int Id { get; set; }
        [Required]
        public string Name { get; set; } = string.Empty;
        public DateTime? DateCreated { get; set; }
        public bool Complete { get; set; }
        public DateTime? DateCompleted { get; set; }

        [ForeignKey("User")]
        public int UserId { get; set; }
        public User User { get; set; } = default!;
    }
}
